src/components/map/CreateGroupCard.tsx

```diff
@@ -86,14 +77,40 @@ const CreateGroupCard = ({ onClose }: CreateGroupCardProps) => {
 
   // Creates a group with the selected users
   const handleCreateGroup = () => {
-    Alert.alert(
-      'TODO: create a group with these users: ',
-      JSON.stringify(
-        selectedUsers.map(user => user.firstName + ' ' + user.lastName)
-      )
-    );
-    // TODO: close if group creation is successful; otherwise, don't
-    onClose();
+    if (!userDocument) return;
+    if (selectedUsers.length === 0) return Alert.alert('No users selected!');
+
+    // create a group object
+    const groupObject: Group = {
+      name: 'Group_' + Math.floor(Math.random() * 1000),
+      members: [userDocument._id],
+      invitedMembers: selectedUsers.map(user => user._id),
```

Comment:
does map return an empty array if none are there or if invited members is undefined?